lib/validate.rb

```diff
@@ -0,0 +1,70 @@
+class Validate
+
+    attr_accessor :message
+    def initialize
+        @message = {"error": {}}
+    end
+
+    def validate_startgame(game_id, datastore)
+        if datastore.has_key?(game_id)
+            @message[:error].store("game", "game #{game_id} already exists")
+            return false
+        end
+        return true
+    end
+
+    def validate_available_moves(game_id, datastore)
+        if !datastore.has_key?(game_id)
+            @message[:error].store("game", "game #{game_id} hasn't yet started")
+            return false
+        else
+            return true
+        end
+    end
+
+    def validate_turns(game_id, datastore)
+        if !datastore.has_key?(game_id)
+            @message[:error].store("turn", "game #{game_id} not started")
+            return false
+        else
+            return true
+        end
```

Comment:
Personally I would have flipped the branches so that I don't have to deal with an extra negation. For example,

```rb
        if datastore.has_key?(game_id)
            true
        else
            @message[:error].store("turn", "game #{game_id} not started")
            false
        end
```